Description: Fix printf format in usbms.c; add 'z' for grub_size_t
Author: sean finney <seanius@debian.org>
Author: Colin Watson <cjwatson@debian.org>
Bug-Debian: http://bugs.debian.org/584474
Forwarded: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=584474#10
Last-Update: 2010-06-04

Index: b/disk/usbms.c
===================================================================
--- a/disk/usbms.c
+++ b/disk/usbms.c
@@ -312,7 +312,7 @@
       grub_dprintf ("usb", "buf:\n");
       if (size <= 64)
         for (i=0; i<size; i++)
-          grub_dprintf ("usb", "0x%02x: 0x%02x\n", i, buf[i]);
+          grub_dprintf ("usb", "0x%02zx: 0x%02x\n", i, buf[i]);
       else
           grub_dprintf ("usb", "Too much data for debug print...\n");
     }
@@ -330,7 +330,7 @@
       /* Debug print of sent data. */
       if (size <= 256)
         for (i=0; i<size; i++)
-          grub_dprintf ("usb", "0x%02x: 0x%02x\n", i, buf[i]);
+          grub_dprintf ("usb", "0x%02zx: 0x%02x\n", i, buf[i]);
       else
           grub_dprintf ("usb", "Too much data for debug print...\n");
     }
Index: b/kern/misc.c
===================================================================
--- a/kern/misc.c
+++ b/kern/misc.c
@@ -688,6 +688,7 @@
 	  int n;
 	  int longfmt = 0;
 	  int longlongfmt = 0;
+	  int sizetfmt = 0;
 	  int unsig = 0;
 
 	  if (*fmt && *fmt =='-')
@@ -740,6 +741,11 @@
 		  c = *fmt++;
 		}
 	    }
+	  else if (c == 'z')
+	    {
+	      sizetfmt = 1;
+	      c = *fmt++;
+	    }
 
 	  switch (c)
 	    {
@@ -770,6 +776,11 @@
 		  long l = va_arg (args, long);
 		  grub_lltoa (tmp, c, l);
 		}
+	      else if (sizetfmt)
+		{
+		  grub_size_t sz = va_arg (args, grub_size_t);
+		  grub_lltoa (tmp, c, sz);
+		}
 	      else if (unsig)
 		{
 		  unsigned u = va_arg (args, unsigned);
